budgeteer-web-interface/src/main/java/org/wickedsource/budgeteer/service/security/configuration/SecurityConfig.java

```diff
@@ -0,0 +1,41 @@
+package org.wickedsource.budgeteer.service.security.configuration;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.web.context.SecurityContextPersistenceFilter;
+import org.wickedsource.budgeteer.service.user.User;
+import org.wickedsource.budgeteer.web.components.security.NeedsLogin;
+import org.wickedsource.budgeteer.service.user.UserService;
+
+/**
+ * A configuration to set up spring boot security. Only used for authorization. Authentication
+ * is performed via the {@link NeedsLogin} annotation and the {@link UserService}.
+ *
+ * @see org.wickedsource.budgeteer.web.BudgeteerSession#login(User)
+ * @see UserService#login(String, String)
+ */
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        // @formatter:off
+        http
+                .headers()
+                    .frameOptions()
+                        .sameOrigin() // allow iframes on the same domain (required for editing contracts etc.)
+                .and()
+                .csrf()
+                    .disable() // disable CSRF
```

Comment:
Is there a reason why you disabled the CSRF-protection?

Reply:
I mainly disabled it because I didn't want to mess around with the wicket forms used on the pages too much, but I can enable it of course.

Reply:
I just mean: disabling default security featues is mostly not a good idea if you don't have good reasons why you do so.. just wanted to point that out. What you do about it is your responsibility :)

Reply:
I will try to enable it, it would surely improve security.

Thanks for the feedback! :)